### DISCUSSION TITLE
x86 folder inside x64 distribution/installation

### DISCUSSION THREAD

**Discussion by Vasilich** [winsiderss/systeminformer#2347] [Questions / Answers] [answered]
What is the purpose of having x86 foldr inside x64 installation? I thought that if i use x64Windows and install x64 SI, then i don't need x86 binaries. But these are even get updated with update function. Am I wrong? Or is it just the current way the installer/updater works?
The x86 binaries are the same in the zip archive in i386 and in amd64/x86 folders.

**Comment by dmex** [comment, marked as answer]
The Native API doesn't natively support the WoW64 (Windows 32-bit on Windows 64-bit) subsystem because of differences in the architecture and memory addressing between 32-bit and 64-bit.

A large amount of structures and system objects only exist in user mode where the WoW64 subsystem has a thunking layer for emulating 32-bit system objects on 64bit, such as the Thread Environment Block (TEB) and the Process Environment Block (PEB) while also intercepting 32-bit functions to convert objects/data from 32-bit to 64-bit before switching to long mode (native 64-bit) to continue execution.

It's easier to query 32-bit objects from 32-bit processes by compiling the exact same 64-bit code for 32-bit and querying from WoW64 instead of trying to support the code required to query the information from 64-bit and all the architectural and system changes between different processors and versions of Windows.

<img width="588" alt="pebteb" src="https://github.com/user-attachments/assets/7d6de620-c778-4389-b498-3b38ea7624fb" />


Task Manager and Process Explorer do the same thing but one secretly extracts the 32-bit binaries from resources while the other uses binaries from C:\Windows\SysWOW64\
